include/ck/utility/inner_product.hpp

```diff
@@ -76,14 +76,16 @@ template <>
 __device__ void inner_product<half2_t, half2_t, float>(const half2_t& a, const half2_t& b, float& c)
 {
 #if defined(CK_USE_AMD_V_DOT2_F32_F16)
-#if CK_USE_AMD_INNER_PRODUCT_INLINE_ASM
+#if CK_USE_AMD_V_DOT_INLINE_ASM
+    // Use 3 x s_nop to avoid hazard (mi200 cdna2 isa)
     asm volatile("\n \
             v_dot2_f32_f16 %0, %1, %2, %0\n \
+            s_nop 2 \n \
```

Comment:
In above comment you said one need to repeat nop 3 times. Isn't this `2` a mistake here or the comment is outdated?

Reply:
Added comment. s_nop 2 is equal to 3x s_nop